libros/packages/add_book_to_library.py

```diff
@@ -0,0 +1,82 @@
+#! /usr/bin/env python3
+
+import pymongo
+import readchar
+from colorama import Fore, Style
+
+from packages.get_dewey_category import get_category
+from packages.get_dewey_subcategory import get_subcategory
+from packages.get_book_from_oclc import get_book
+
+""" Adds a single book to the database
+
+Usage:
+
+    python3 add_book_to_library.py
+"""
+def add_book(barcode):
+    client = pymongo.MongoClient('10.56.1.30', 27017)
+    libaro_db = client["libaro"]
+    books = libaro_db['books']
+
+    book_data = get_book(barcode)
+
+    existing_copy = books.find_one({'isbn': barcode})
+
+    add_book = True
+
+    if (existing_copy):
+        print(Fore.RED + 'This book already exists in the library.\n')
+        print('Press' + Fore.GREEN + ' 1 ' + Style.RESET_ALL + 'to add another copy or any other key to return to the main menu\n')
+    
+        add_new_book = readchar.readkey()
+
+        if not (add_new_book == '1'):
+            add_book = False
+
+    if add_book:
+        try:
+            # Get book
+            book = book_data[1].attrib
+
+            # Get dewey number
+            dewey = book_data[5][0][1].attrib['sfa']
+        except:
+            print(Fore.RED + 'Unable to parse book data')
+
+        try:
+            _, cat_name, cat_colour = get_category(dewey)
+        except:
+            print(Fore.RED + 'Unable to retrieve category')
+        try:
+            subcat_name = get_subcategory(dewey)
+        except:
+            print(Fore.RED + 'Unable to retreive subcategory')
+
+        data = {
+            'isbn': barcode,
```

Comment:
Probably a silly question, but could there conceivably be multiple copies of the same book?
Does the barcode uniquely ID the book or would multiple copies have the same barcode?

Reply:
The barcode is the book title/edition etc... if you scan the barcode multiple times, you end up with multiple copies of the same book. Which works well for us as we do have multiple copies of a couple of books.